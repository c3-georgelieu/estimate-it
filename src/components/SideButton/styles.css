.side-button {
  width: 100%;
  text-align: center;
  padding-bottom: 4px;
}

.side-button button {
  width: 38px;
  padding-left: initial;
  padding-right: initial;
}

.side-button::after {
  content: attr(data-title);
  display: block;
  position: relative;
  font-size: 10px;
  transition: transform .4s 1s;
  font-variant: all-petite-caps;
}

.side-button[data-color='primary'] { color: #007bff; }
.side-button[data-color='danger'] { color: #dc3545; }
.side-button[data-color='secondary'] { color: #6c757d; }
.side-button[data-color='warning'] { color: #ffc107; }
.side-button[data-color='success'] { color: #28a745; }

.estimate aside .side-button:nth-child(1)::after { transform: translate(calc(1 * var(--offset))); }
.estimate aside .side-button:nth-child(2)::after { transform: translate(calc(2 * var(--offset))); }
.estimate aside .side-button:nth-child(3)::after { transform: translate(calc(3 * var(--offset))); }
.estimate aside .side-button:nth-child(4)::after { transform: translate(calc(4 * var(--offset))); }
.estimate aside .side-button:nth-child(5)::after { transform: translate(calc(5 * var(--offset))); }
.estimate aside .side-button:nth-child(6)::after { transform: translate(calc(6 * var(--offset))); }
.estimate aside .side-button:nth-child(7)::after { transform: translate(calc(7 * var(--offset))); }
.estimate aside .side-button:nth-child(8)::after { transform: translate(calc(8 * var(--offset))); }
.estimate aside .side-button:nth-child(9)::after { transform: translate(calc(9 * var(--offset))); }

.estimate aside:hover .side-button::after {
  transform: translate(0);
  transition: transform .4s;
}
