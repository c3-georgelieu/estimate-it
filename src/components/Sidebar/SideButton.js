import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import FA from 'react-fontawesome'
import { Button } from 'reactstrap'

export const SideButton = ({
  title,
  name,
  color,
  onClick,
  link,
}) => (
  <div className="side-button">
    {link ? (
      <Link to={link}>
        <Button
          outline
          color={color}
        >
          <FA name={name} />
        </Button>
      </Link>
    ) : (
      <Button
        outline
        color={color}
        onClick={onClick}
      >
        <FA name={name} />
      </Button>
    )}
    <div className={`title text-color-${color}`}>
      {title}
    </div>
  </div>
)

SideButton.propTypes = {
  title: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  color: PropTypes.string,
  onClick: PropTypes.func,
  link: PropTypes.string,
}

SideButton.defaultProps = {
  color: 'primary',
  onClick: undefined,
  link: undefined,
}

export default SideButton
