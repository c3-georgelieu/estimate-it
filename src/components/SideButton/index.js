import { connect } from 'react-redux'
import SideButton from './SideButton'
import { redirect } from '../../redux/actions/async'
import './styles.css'

export default connect(null, ({ redirect }))(SideButton)
