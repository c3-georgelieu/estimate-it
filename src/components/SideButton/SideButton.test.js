import React from 'react'
import renderer from 'react-test-renderer'
import SideButton from './SideButton'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <SideButton
        title="New"
        name="file"
        link="/estimate/new"
        redirect={_ => null}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
