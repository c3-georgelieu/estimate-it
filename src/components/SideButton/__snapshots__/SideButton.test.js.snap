// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="side-button"
  data-color="primary"
  data-title="New"
>
  <button
    className="btn btn-outline-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden={true}
      className="fa fa-file"
    />
  </button>
</div>
`;
